testing/trino-tests/src/test/java/io/trino/execution/TestViewWithTableFunction.java

```diff
@@ -0,0 +1,91 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.trino.execution;
+
+import io.trino.Session;
+import io.trino.plugin.base.security.AllowAllSystemAccessControl;
+import io.trino.plugin.jdbc.JdbcPlugin;
+import io.trino.plugin.jdbc.TestingH2JdbcModule;
+import io.trino.plugin.memory.MemoryPlugin;
+import io.trino.spi.security.Identity;
+import io.trino.spi.security.SystemAccessControl;
+import io.trino.testing.AbstractTestQueryFramework;
+import io.trino.testing.DistributedQueryRunner;
+import io.trino.testing.QueryRunner;
+import io.trino.testing.TestingAccessControlManager;
+import io.trino.testing.TestingAccessControlManager.TestingPrivilege;
+import org.testng.annotations.Test;
+
+import java.util.Map;
+
+import static io.airlift.testing.Closeables.closeAllSuppress;
+import static io.trino.SessionTestUtils.TEST_SESSION;
+import static io.trino.testing.TestingAccessControlManager.TestingPrivilegeType.GRANT_EXECUTE_FUNCTION;
+import static org.assertj.core.api.Assertions.assertThat;
+
+public class TestViewWithTableFunction
+        extends AbstractTestQueryFramework
+{
+    @Override
+    protected QueryRunner createQueryRunner()
+            throws Exception
+    {
+        // TODO (https://github.com/trinodb/trino/issues/12833) switch to FileBasedSystemAccessControl once it supports access rules for table functions
```

Comment:
https://github.com/trinodb/trino/issues/12833

Reply:
FYI @huberty89 @ssheikin

Reply:
I think testing `FileBasedSystemAccessControl` is separate thing. Here make sure that engine works, there will check if plugin works. So I would remove the TODO from here.